composer.json

```diff
@@ -72,6 +72,60 @@
     "bin": [
         "php-cs-fixer"
     ],
+    "scripts": {
+        "cs:check": "php php-cs-fixer fix --dry-run",
+        "cs:diff": "git diff HEAD --name-only --diff-filter=ACMR -- '*.php' | xargs -r php php-cs-fixer fix --dry-run --diff",
+        "cs:diff-fix": "git diff HEAD --name-only --diff-filter=ACMR -- '*.php' | xargs -r php php-cs-fixer fix",
+        "cs:fix": "php php-cs-fixer fix",
+        "dev-tools:check": [
+            "./dev-tools/check_file_permissions.sh",
+            "./dev-tools/check_trailing_spaces.sh"
+        ],
+        "dev-tools:docs": "php dev-tools/doc.php",
+        "dev-tools:install": "composer -d dev-tools install",
+        "mess-detector": "php dev-tools/vendor/bin/phpmd . ansi phpmd.xml",
+        "phpstan": "php -d memory_limit=256M dev-tools/vendor/bin/phpstan analyse",
+        "qa": "@quality-assurance",
+        "quality-assurance": [
+            "Composer\\Config::disableProcessTimeout",
+            "@cs:check",
+            "@sa",
+            "@test"
```

Comment:
I am not sure if `@test` should be here or only `@test:diff` or `@test:branch`. Running all tests takes long and I would like to use `c qa` on daily basis. Let me know what you think 🙂

Reply:
Since `cs:check` was moved to `sa` script I believe `test` is OK here. Normally one would like to run `c sa` (PHPStan + Fixer) during development and probably `c test` before push.